ansible_collections/arista/avd/plugins/action/validate.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import copy
+from ansible.plugins.action import ActionBase
+from ansible.errors import AnsibleActionFail
+from ansible_collections.arista.avd.plugins.plugin_utils.schema.avdschema import AristaAvdError, AvdSchema
+from ansible.utils.display import Display
+
+
+VALID_CONVERSION_MODES = ["disabled", "warning", "message", "debug"]
+VALID_VALIDATION_MODES = ["disabled", "error", "warning", "message", "debug"]
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = {}
+
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        # ########
+        # VALIDATE ARGUMENTS
+        if self._task.args and "schema" in self._task.args:
+            schema = self._task.args["schema"]
+            if not isinstance(schema, dict):
+                raise AnsibleActionFail("The argument 'schema' is not a dict")
+        else:
+            raise AnsibleActionFail("The argument 'schema' must be set")
+
+        conversion_mode = self._task.args.get("conversion_mode", "debug")
+        if not isinstance(conversion_mode, str):
+            raise AnsibleActionFail("The argument 'conversion_mode' must be a string")
+        if conversion_mode not in VALID_CONVERSION_MODES:
+            raise AnsibleActionFail(f"Invalid value '{conversion_mode}' for the argument 'conversion_mode'."
+                                    f"Must be one of {VALID_CONVERSION_MODES}")
+
+        validation_mode = self._task.args.get("validation_mode", "warning")
+        if not isinstance(validation_mode, str):
+            raise AnsibleActionFail("The argument 'validation_mode' must be a string")
+        if validation_mode not in VALID_VALIDATION_MODES:
+            raise AnsibleActionFail(f"Invalid value '{validation_mode}' for the argument 'validation_mode'."
+                                    f"Must be one of {VALID_VALIDATION_MODES}")
+
+        # ########
+        # READ ANSIBLE VARIABLES AND PERFORM TEMPLATING TO SUPPORT INLINE JINJA
+        hostname = task_vars['inventory_hostname']
+        data = {}
+        for var in task_vars:
+            if str(var).startswith(("ansible", "molecule", "hostvars", "vars")):
```

Comment:
probably not. The main purpose is just to avoid these variables in the data, since they are not always regular variables, but also objects if varying classes.